### DISCUSSION TITLE
WinForms - Disabling Visibility Events

### DISCUSSION THREAD

**Discussion by adsonvinicius** [cefsharp/CefSharp#3394] [General]
I want to know if is possible to create a feature which allows disable Visibility Events. For example I'm struggling to figure out a way to avoid the browser knows the Forms app is minimized, because the content stops updating when is minimized

**Comment by amaitland** [comment]
Are you using WinForms or WPF?

**Reply by adsonvinicius** [reply]
WinForms

**Reply by amaitland** [reply]
You can P/Invoke SetWindowPos to resume rendering after the form is minimized 

The following is the C++ used, you can easily translate it to c#
```
SetWindowPos(browserHwnd, NULL, 0, 0, width, height, SWP_NOZORDER)
```

You can get the hwnd using the following. 
http://cefsharp.github.io/api/86.0.x/html/M_CefSharp_IBrowserHost_GetWindowHandle.htm
http://cefsharp.github.io/api/86.0.x/html/M_CefSharp_WebBrowserExtensions_GetBrowserHost.htm

**Reply by adsonvinicius** [reply]
So do you believe if a page has a `document.hidden` property checked by an interval, this `document.hidden` will be true even if form is minimized?

I am just checking if I could explain what exactly I want before understand that hook and translate to C#

**Reply by amaitland** [reply]
That's exactly the same call used internally when the browser becomes visible after minimize. 

https://github.com/cefsharp/CefSharp/blob/cefsharp/87/CefSharp.Core.Runtime/ManagedCefBrowserAdapter.cpp#L180

**Reply by adsonvinicius** [reply]
I got your point. I will try that. 

Thanks.

**Reply by kevinvenclovas** [reply]
@adsonvinicius  Did you get an solution? I also need to run my form when its minimized.

**Reply by adsonvinicius** [reply]
@kevinvenclovas I decided to hide some areas of the website. In my case the problem was the High CPU level due to many changes on page by second, so I reduced and let the form shown. Technically the solution above (calling SetWindowPos) works fine.
